docs/source/developer_guides/custom_models.md

```diff
@@ -238,3 +238,66 @@ peft_model.print_trainable_parameters()
 ```python
 print(peft_model.targeted_module_names)
 ```
+
+## Unsupported module types
+
+In PEFT in general, methods like LoRA only work if the target modules are supported by PEFT. For example, it's possible to apply LoRA to `nn.Linear` and `nn.Conv2d` layers, but not, for instance, to `nn.LSTM`. If you find that the class of layer you want to apply PEFT to is not supported, feel free to open a [PEFT issue on GitHub](https://github.com/huggingface/peft/issues) and request the feature. Maintainers will implement this or guide you on how to implement this yourself if the demand for this module type is sufficiently high.
+
+### Experimental support for dynamic dispatch of custom modules in LoRA
+
+> [!WARNING]
+> This feature is experimental and subject to change, depending on its reception by the community. If we find that there is a big demand, we will introduce a public, stable API.
+
+For LoRA specifically, PEFT supports an experimental API for custom module types. Let's assume that you have a LoRA implementation for LSTMs. Normally, you would not be able to tell PEFT to use it, even if it would theoretically work with PEFT. With dynamic dispatch of custom layers, you can, however, make this work.
```

Comment:
```suggestion
PEFT supports an experimental API for custom module types for LoRA. Let's assume you have a LoRA implementation for LSTMs. Normally, you would not be able to tell PEFT to use it, even if it would theoretically work with PEFT. However, this is possible with dynamic dispatch of custom layers.
```